src/features/navigator/NarrativeMetadata.tsx

```diff
@@ -0,0 +1,185 @@
+// <NarrativeMetadata />
+import { FC } from 'react';
+import { useAppSelector } from '../../common/hooks';
+import {
+  Cell,
+  DataObject,
+  isCodeCell,
+  isKBaseAppCell,
+  isKBaseCodeTypeCell,
+  isKBaseDataCell,
+  isMarkdownCell,
+  NarrativeDoc,
+} from '../../common/types/NarrativeDoc';
+import { corruptCellError } from './common';
+import { useUsers } from './hooks';
+import { users } from './navigatorSlice';
+import classes from './Navigator.module.scss';
+
+const cellType = (cell: Cell, index: number) => {
+  if (isMarkdownCell(cell)) {
+    return 'markdown';
+  }
+  if (!isCodeCell(cell)) {
+    corruptCellError(cell, index);
+    return 'corrupt';
+  }
+  const kbaseData = cell.metadata.kbase;
+  if (!kbaseData) {
+    corruptCellError(cell, index);
+    return 'corrupt';
+  }
+  if (isKBaseAppCell(kbaseData)) {
+    return 'kbase_app';
+  }
+  if (isKBaseDataCell(kbaseData)) {
+    return 'data';
+  }
+  if (isKBaseCodeTypeCell(kbaseData)) {
+    return 'code_cell';
+  }
+  // Other cells are not counted.
+  return 'other';
+};
+
+const countCellTypes = (cells: Cell[]) => {
+  const defaults = {
+    markdown: 0,
+    code_cell: 0,
+    data: 0,
+    kbase_app: 0,
+  };
+  return cells.reduce(
+    (acc: Record<string, number>, cell: Cell, index: number) => {
+      acc[cellType(cell, index)] += 1;
+      return acc;
+    },
+    defaults
+  );
+};
+
+const readableDate = (date: string | number): string => {
```

Comment:
Could probably be pulled out into a higher level util file in the future but might need a few more config options